pkg/mimir/server_service_test.go

```diff
@@ -69,3 +57,13 @@ func TestServerStopViaStop(t *testing.T) {
 	require.Error(t, s.AwaitTerminated(context.Background()))
 	require.Equal(t, services.Failed, s.State())
 }
+
+func getServerCfg(t *testing.T) server.Config {
```

Comment:
This is quite similar to `getServerConfig()`. Can you just reuse `getServerConfig()` and set the registerer there?